keyboards/mlego/m20/m20.c

```diff
@@ -0,0 +1,205 @@
+        led_rse(true);
+    }
+
+}
+
+#ifdef ENCODER_ENABLE
+
+#    define MEDIA_KEY_DELAY 10
+
+void my_encoders(const uint8_t index, const bool clockwise) {
+    if (index == 0) { /* First encoder */
+#ifdef RGBLIGHT_ENABLE
+        if (IS_LAYER_ON(_LWR)) {
+            if (clockwise) {
+                rgblight_decrease_val_noeeprom();
+            } else {
+                rgblight_increase_val_noeeprom();
+            }
+        } else if (IS_LAYER_ON(_RSE)) {
+            if (clockwise) {
+                rgblight_decrease_hue_noeeprom();
+            } else {
+                rgblight_increase_hue_noeeprom();
+            }
+
+        } else {
+#endif
+            if (clockwise) {
+                tap_code_delay(KC_VOLD, MEDIA_KEY_DELAY);
+            } else {
+                tap_code_delay(KC_VOLU, MEDIA_KEY_DELAY);
+            }
+#ifdef RGBLIGHT_ENABLE
+        }
+#endif
+    }
+}
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) { return false; }
+    my_encoders(index, clockwise);
+    return false;
+}
+
+#endif
+
+#ifdef OLED_ENABLE
+
+void init_timer(void){
+   oled_logo_timer = timer_read32();
+};
+
+void user_oled_magic(void) {
+    // Host Keyboard Layer Status
+    oled_write_P(PSTR("Layer: "), false);
+
+    switch (get_highest_layer(layer_state)) {
+        case _NUM:
+            oled_write_P(PSTR("Default\n"), false);
+            break;
+        case _LWR:
+            oled_write_P(PSTR("Lower\n"), false);
+            break;
+        case _RSE:
+            oled_write_P(PSTR("Raise\n"), false);
+            break;
+        case _ADJ:
+            oled_write_P(PSTR("ADJ\n"), false);
+            break;
+        default:
+            // Or use the write_ln shortcut over adding '\n' to the end of your string
+            oled_write_ln_P(PSTR("Undefined"), false);
+    }
+
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("Lower ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("Raise ") : PSTR("    "), false);
+#ifdef WPM_ENABLE
+    oled_write_P(PSTR("\nwpm: "), false);
+    uint8_t wpm = get_current_wpm();
+    oled_write_P(wpm != 0 ? get_u8_str(wpm,' ') : PSTR("   "), false);
+#endif
+}
+
+void render_logo(void) {
+    oled_write_P(m65_logo, false);
+}
+
+void clear_screen(void) {
+    if (clear_logo){
+      for (uint8_t i = 0; i < OLED_DISPLAY_HEIGHT; ++i) {
+        for (uint8_t j = 0; j < OLED_DISPLAY_WIDTH; ++j) {
+          oled_write_raw_byte(0x0, i*OLED_DISPLAY_WIDTH + j);
+        }
+      }
+      clear_logo = false;
+    }
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_180;
+}
+
+#    define SHOW_LOGO 5000
+bool oled_task_kb(void) {
+    if (!oled_task_user()) { return false; }
+    if ((timer_elapsed32(oled_logo_timer) < SHOW_LOGO)){
+        render_logo();
+    }else{
+      clear_screen();
```

Comment:
why not just call oled clear?

Reply:
is this some new feature? is the same code i use for mlego/m65

Reply:
Nope,`oled_clear()` has been in there for a long while.  But I've been growing more familiar with the oled code, as well.

Reply:
ok now i remember, oled_clear() simply did not remove the logo image.
i even did a call to oled_render() after and no luck... I used that brutal method like was 1990 again